packages/cms-cli/commands/watch.js

```diff
@@ -28,13 +28,125 @@ const {
 } = require('../lib/usageTracking');
 
 const COMMAND_NAME = 'watch';
+const DESCRIPTION =
+  'Watch a directory on your computer for changes and upload the changed files to the HubSpot CMS';
 
-function configureWatchCommand(program) {
+const action = async ({ src, dest }, command = {}) => {
+  setLogLevel(command);
+  logDebugInfo(command);
+  const {
+    config: configPath,
+    remove,
+    initialUpload,
+    disableInitial,
+    notify,
+  } = command;
+  loadConfig(configPath);
+  checkAndWarnGitInclusion();
+
+  if (disableInitial) {
+    logger.info(
+      'Passing the "--disable-initial" option is no longer necessary. Running "hs watch" no longer uploads the watched directory by default.'
+    );
+  } else {
+    logger.warn(
+      `The "watch" command no longer uploads the watched directory by default. The directory "${src}" was not uploaded.`
+    );
+    logger.warn(
+      'To upload the directory run "hs upload" or add the "--initial-upload" option when running "hs watch".'
+    );
+  }
+
+  if (
+    !(
+      validateConfig() &&
+      (await validatePortal(command)) &&
+      validateMode(command)
+    )
+  ) {
+    process.exit(1);
+  }
+
+  const portalId = getPortalId(command);
+  const mode = getMode(command);
+
+  const absoluteSrcPath = path.resolve(getCwd(), src);
+  try {
+    const stats = fs.statSync(absoluteSrcPath);
+    if (!stats.isDirectory()) {
+      logger.log(`The "${src}" is not a path to a directory`);
+      return;
+    }
+  } catch (e) {
+    logger.log(`The "${src}" is not a path to a directory`);
+    return;
+  }
+
+  if (!dest) {
+    logger.log('A destination directory needs to be passed');
+    return;
+  }
+
+  trackCommandUsage(COMMAND_NAME, { mode }, portalId);
+  watch(portalId, absoluteSrcPath, dest, {
+    mode,
+    cwd: getCwd(),
+    remove,
+    disableInitial: initialUpload ? false : true,
+    notify,
+  });
+};
+
+// Yargs Configuration
+const command = `${COMMAND_NAME} <src> <dest>`;
+const describe = DESCRIPTION;
+const builder = yargs => {
+  addConfigOptions(yargs, true);
+  addPortalOptions(yargs, true);
+  addLoggerOptions(yargs, true);
+  addModeOptions(yargs, { write: true }, true);
+
+  yargs.positional('src', {
+    describe: 'Local folder to watch',
+    type: 'string',
+    demand: true,
+  });
+  yargs.positional('dest', {
+    describe: 'Remote folder to upload to',
+    type: 'string',
+    demand: true,
+  });
+  yargs.option('remove', {
+    alias: 'r',
+    describe: 'remove remote files when removed locally',
+    type: 'boolean',
+  });
+  yargs.option('initial-upload', {
+    alias: 'i',
+    describe: 'upload directory before watching for updates',
+    type: 'boolean',
+  });
+  yargs.option('disable-initial', {
+    alias: 'd',
+    describe: 'retrieve most recent log only',
+    type: 'disable initial upload of watched directory (default)',
+  });
+  yargs.option('notify', {
```

Comment:
couldn't find a way to do this command like commander, where it shows `--notify <path/to/file>` in the name so ended up just adding it in the description

Reply:
I think we should remove `<` and `>` in the description since the documentation is outside the context of the structure of an option or argument. Instead I think an example might be best like `/tmp/hs-watch` or something.

Reply:
Understood, will change.